modules/friendlytag.js

```diff
@@ -147,18 +160,59 @@ Twinkle.tag.callback = function friendlytagCallback() {
 	Window.display();
 
 	if (Twinkle.tag.mode === "article") {
+
+		Twinkle.tag.alreadyPresentTags = [];
+
+		if (Twinkle.tag.untaggable) {
+			// Look for existing maintenance tags in the lead section and put them in array
+
+			// All tags are HTML table elements that are direct children of .mw-parser-output,
+			// except when they are within {{multiple issues}}
+			$('.mw-parser-output').children().each( function parsehtml(i,e) {
```

Comment:
I'm kind of thinking out loud here, but is there another way to do this?  I was thinking initially that one advantage of going after `.mw-parser-output` is that you avoid having to find all the damn redirects, but that's still required to figure out what to uncheck, right?

I had thought to suggest an API call to query; `extracts` doesn't provide you wikitext, but `revisions` does.  And we can specify just the first section with `rvsection`.  If we grabbed that text, we could, for every `{{`:
1. Check if it's one of our known `tag`s
2. If not, check if it's a redirect to one of our known `tag`s via `query`

The reason why I bring it up is that the current way of checking redirects (removing everything found in linkshere) seems more dangerous and prone to mistakes (see related comment about other templates that use classes with `box-`).

Reply:
Wouldn't that a be a lot of api calls? Even a single api call requires some time during which the interface is a blank screen.  

I don't think the present procedure is dangerous or error-prone. Note that the removal regex doesn't have the `/g` flag - only the first occurrence of the template will be removed (which is likely to be the one in the lead section, if at all there are multiple). And after one redirect has matched, the remaining wouldn't even be checked. 

Removal of a tag takes place only when: 
- An HTML element bearing the class `box-Tag_name` was found in the lead section
- The editor deselected `{{Tag name}}` from the interface, marking it for removal
- The wikitext corresponding to the the template or its redirect was found 

All three of these wouldn't occur accidentally. #1 to go wrong would require an admin/TE vandal in most cases. 

In any case, the regexes are strong enough, not even allowing caps variations. So I don't see how anything that shouldn't have been removed would ever get removed.

Reply:
Was writing a thing, but your latest code push completely nullifies my argument anyway, methinks.